### DISCUSSION TITLE
Questions from an upstream developer about updating a formula in core

### DISCUSSION THREAD

**Discussion by sidney** [Homebrew/discussions#302] [Everyday usage] [answered]
I'm one of the upstream developers for the package that is installed by core formula "enigma", currently at version 1.21.
We are preparing an alpha of a new version release. Already, the existing formula will no longer build from HEAD due to changes we have made in master in dependencies from SDL 1.2 to SDL 2.

Should we submit a pull request that updates the formula so it still uses our 1.21 release as the stable and has different dependency lists for stable and HEAD? Is there a way of having the formula reference our 1.30-alpha when it is released, which should be this month?

Or should we just hold off on doing any pull request until we officially release Enigma 1.30?

Also, the enigma.rb in core builds an Enigma.app bundle that depends on various shared libraries that are assumed to be installed by Homebrew. I want to write a formula named something like "enigma-distribute" which will depend on enigma and convert Enigma.app to have dylib files bundled in it with linkage fixed up to work, so that it can be downloaded as a standalone application. Would that formula be acceptable for core? Having the bottles would mean that we don't have to have every supported version of MacOS to get builds on them.

Thanks

**Comment by gromgit** [comment, marked as answer]
> Or should we just hold off on doing any pull request until we officially release Enigma 1.30?

From what I've seen, that would be the desired option, as Homebrew core only accepts stable releases. Of course, you can host pre-release formulae on your own tap, for testing purposes: https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap

> I want to write a formula named something like "enigma-distribute" which will depend on enigma and convert Enigma.app to have dylib files bundled in it with linkage fixed up to work, so that it can be downloaded as a standalone application. Would that formula be acceptable for core?

Unlikely, since that sounds like a tool with a limited audience. You can host it on your own tap, of course.
